src/inputs/ObjectSearchCreate.tsx

```diff
@@ -76,36 +77,30 @@ class ObjectSearchCreate extends Component<IObjectSearchCreateProps> {
     if (onAddNewToggle) { onAddNewToggle(false); }
   }
 
-  public render () {
-    const {
-      decoratorOptions,
-      fieldConfig,
-      formManager,
-    } = this.injected
-    , className = cx(
-      CX_PREFIX_SEARCH_CREATE,
-      {[`${CX_PREFIX_SEARCH_CREATE}-create`]: this.isAddingNew.isTrue },
+  private renderAddNew () {
+    const { fieldConfig, formManager } = this.injected;
+
+    return (
+      <>
+        <NestedFieldSet
+          fieldSet={this.fieldConfig.createFields}
+          formManager={formManager}
+          id={fieldConfig.field}
+          label={this.fieldConfig.label}
```

Comment:
This seems messy. No pulling things out of fieldConfigs, and no consistency of `this.fieldConfig` vs { fieldConfig } = this.props;` above. I'd be fine with us pulling all out of this.fieldConfig, or just using this.fieldConfig all over, but mixing seems bad.

Reply:
Actually I just realized this is a whitespace diff so ignore 😛

Reply:
@rkuykendall Well, I didn't change any of the props, I simply split the Add and the Search rendering pieces for easier reading of the `render()` function. + to have a proper main `className` on the root element of that component.